src/World.cpp

```diff
@@ -1057,16 +1064,20 @@ void cWorld::Tick(std::chrono::milliseconds a_Dt, std::chrono::milliseconds a_La
 
 	TickWeather(static_cast<float>(a_Dt.count()));
 
-	if (m_WorldAge - m_LastSave > std::chrono::minutes(5))  // Save each 5 minutes
-	{
-		SaveAllChunks();
-	}
-
-	if (m_WorldAge - m_LastUnload > std::chrono::seconds(10))  // Unload every 10 seconds
+	if (m_WorldAge - m_LastChunkCheck > std::chrono::seconds(10))  // Unload every 10 seconds
 	{
 		UnloadUnusedChunks();
+		// Save each 5 minutes
+		if (m_WorldAge - m_LastSave > std::chrono::minutes(5))
+		{
+			SaveAllChunks();
+		}
+		// Save if we have too many dirty unused chunks
+		else if (GetNumUnusedDiryChunks() > m_UnusedDirtyChunksCap)
+		{
+			SaveAllChunks();
+		}
```

Comment:
*Really* minor point, but don't the comments normally go inside the blocks and not between them? :)